homeassistant/components/sensor/recorder.py

```diff
@@ -683,13 +688,13 @@ def validate_statistics(
 
             metadata_unit = metadata[1]["unit_of_measurement"]
             if device_class not in UNIT_CONVERTERS:
-                issue_type = (
-                    "units_changed_can_convert"
-                    if statistics.can_convert_units(metadata_unit, state_unit)
-                    else "units_changed"
-                )
                 if state_unit != metadata_unit:
                     # The unit has changed
+                    issue_type = (
+                        "units_changed_can_convert"
```

Comment:
Let's drop this issue. It's not an issue, everything works as expected

Reply:
We can drop it in a follow-up by also dropping the limitation that we only do unit conversion for certain device classes 👍